src/components/TokenSelect/TokenSelect.tsx

```diff
@@ -85,21 +84,10 @@ const TokenSelect = ({
                   "text-gray-500": !token,
                 }
               )}
-              value={isDefaultOptionDisabled ? token : `…${t("common:select")}`}
-              onChange={(e) => {
-                if (onTokenChange) onTokenChange(e);
-                setIsDefaultOptionDisabled(true);
-              }}
-            >
-              <option disabled={isDefaultOptionDisabled}>
-                …{t("common:select")}
-              </option>
-              {tokens.map((token) => (
-                <option key={token.address} value={token.address}>
-                  {token.symbol}
-                </option>
-              ))}
-            </select>
+              onClick={onTokenChange}
+              >
+              {tokens.find((t)=> t.address === token)?.symbol || isDefaultOptionDisabled ? tokens.find((t)=> t.address === token)?.symbol : `…${t("common:select")}`}
```

Comment:
You can move the ellipsis to after the select, it shows as "...Select" when it should be "Select...", I think the way we had it before was necessary for the option select tag, I guess for button tags it should be done normally.